pycbc/live/snr_optimizer.py

```diff
@@ -0,0 +1,340 @@
+import argparse, numpy
+import time
+import logging
+import types
+from scipy.optimize import differential_evolution, shgo
+from pycbc import (
+    DYN_RANGE_FAC, waveform
+)
+from pycbc.types import zeros
+import pycbc.waveform.bank
+from pycbc.filter import matched_filter_core
+import pycbc.conversions as cv
+
+try:
+    import pyswarms as ps
+except:
+    ps = None
+
+# Set a minimum mass for points tried in optimization allowing for
+# minimal slop relative to the lightest template
+MIN_CPT_MASS = 0.99
+
+# Set a large maximum total mass
+MAX_MTOTAL = 500.
+
+Nfeval = 0
+start_time = time.time()
+
+def callback_func(Xi, convergence=0):
+    global Nfeval
+    logging.info("Currently at %d %s", Nfeval, convergence)
+    if (time.time() - start_time) > 360:
+        return True
+    Nfeval += 1
+
+
+def compute_network_snr_core(v, *argv, raise_err=False):
+    """
+    Compute network SNR as a function over mchirp, eta and two aligned
```

Comment:
IMO this would be a good time to properly document the input values, as 'v' and what has to be in argv are currently mysterious